cerbero/packages/osx/packager.py

```diff
@@ -302,59 +311,70 @@ def _create_product(self, package_type):
+                                 self.empty_packages[package_type], package_type,
+                                 self.config.target_arch, home_folder=self.home_folder)
         distro_path = os.path.join(self.tmp, "Distribution.xml")
         distro.write(distro_path)
         output_file = os.path.join(self.output_dir, self._package_name('.pkg'))
         output_file = os.path.abspath(output_file)
         pb = ProductBuild()
-        pb.create_package(distro_path, output_file, [self.package.relative_path('.')])
+        search_paths = [self.package.relative_path('.')]
+        search_paths += set([os.path.dirname(p) for p in self.packages_paths[package_type].values()])
+        pb.create_package(distro_path, output_file, search_paths)
         return output_file
 
     def _create_packages(self):
+        extra_paths = []
         for p in self.packages:
             m.action(_("Creating package %s ") % p)
-            packager = OSXPackage(self.config, p, self.store)
+            p.pre_package()
+            packager = OSXPackager(self.config, p, self.store)
+            paths = [None, None]
             try:
-                paths = packager.pack(self.output_dir, self.devel, self.force,
+                paths = packager.pack(self.tmp, self.devel, self.force,
                                       self.keep_temp, self.split, self.package.version,
                                       install_dir=self.package.get_install_dir(),
                                       include_dirs=self.include_dirs,
                                       sdk_version=self.package.sdk_version)
                 m.action(_("Package created sucessfully"))
             except EmptyPackageError:
-                paths = [None, None]
+                pass
+            finally:
+                paths = p.post_package(paths, self.output_dir)
 
             if paths[0] is not None:
                 self.packages_paths[PackageType.RUNTIME][p] = paths[0]
             else:
                 self.empty_packages[PackageType.RUNTIME].append(p)
+
             if paths[1] is not None:
                 self.packages_paths[PackageType.DEVEL][p] = paths[1]
             else:
                 self.empty_packages[PackageType.DEVEL].append(p)
 
-    def _create_packages_dmg(self):
-        paths = list(self.packages_paths[PackageType.RUNTIME].values())
+            extra_paths += paths[2:]
+        return extra_paths
+
+    def _create_packages_dmg(self, paths):
         dmg_file = os.path.join(self.output_dir, self._package_name('-packages.dmg'))
 
         m.action(_("Creating image %s ") % dmg_file)
         # create a temporary directory to store packages
-        workdir = os.path.join(self.tmp, "hdidir")
+        workdir = os.path.join(self.tmp, self.package.name)
         os.makedirs(workdir)
         try:
             for p in paths:
-                shutil.copy(p, workdir)
+                shutil.move(p, workdir)
```

Comment:
This will delete the packages from the output dir.
After a package like the fluendo-sdk-installer is packaged you expect to have the following packages:
fluendo-sdk-installer-1.1.0.740-x86_64-packages.dmg 
fluendo-sdk-installer-1.1.0.740-x86_64.pkg
fluendo-sdk-installer-devel-1.1.0.740-x86_64.pkg

Reply:
the dmg archive is just a big zip of the packages. So if you take the dmg archiving option (I put it optional) all those packages are within the dmg archive, it is cumbersome to have them at 2 different places.

Indeed, what the caller expects is what it gets in the post_package() paths array.

Moreover keeping both can create errors during packages creation, one concrete example:
 package signing is done at post_package() stage (we should change this because it should be done during package creation, there are some options on pkgbuild for that, but it's another story). If you create a dmg archive and you let package in place, then the installer package inside the dmg archive will NOT be signed whereas the package outside will be.

That's why to prevent confusion I've simplified the use cases:
- or you take the dmg archive option and this is what you got with all your packages inside the dmg (then you cannot post-process individual packages)
- or you don't take the dmg archive and then you have all your packages in the folder ready for post-processing (fluendo-sdk-installer doesn't use the dmg option for example)